source/dialogs/drun.c

```diff
@@ -547,6 +556,60 @@ static void drun_icon_fetch ( gpointer data, gpointer user_data )
     rofi_view_reload ();
 }
 
+/**
+ * Initialize desktop entry fields.
+ */
+static void drun_collect_entry_fields ( DRunModePrivateData *pd )
+{
+    unsigned int entry_fields_length = 4;
+    pd->matching_entry_fields                     = g_realloc ( pd->matching_entry_fields, entry_fields_length * sizeof ( DRunEntryField ) );
```

Comment:
Is there a reason why this structure is created at runtime?  it looks like it can easily be a static global to the file.
```c
static DRunEntryField matching_entry_fields[DRUN_MATCH_NUM_FIELDS] = {
    { .entry_field_name = "name", .enabled = TRUE, },
   { },
};
```

Reply:
You're right, I got confused, sorry. Not quite used to writing in C.